sensing/autoware_pointcloud_preprocessor/src/downsample_filter/pickup_based_voxel_grid_downsample_filter_node.cpp

```diff
@@ -78,35 +81,28 @@ PickupBasedVoxelGridDownsampleFilterComponent::PickupBasedVoxelGridDownsampleFil
     std::bind(&PickupBasedVoxelGridDownsampleFilterComponent::param_callback, this, _1));
 }
 
-void PickupBasedVoxelGridDownsampleFilterComponent::filter(
-  const PointCloud2ConstPtr & input, [[maybe_unused]] const IndicesPtr & indices,
-  PointCloud2 & output)
-{
-  std::scoped_lock lock(mutex_);
-
-  stop_watch_ptr_->toc("processing_time", true);
-
-  using VoxelKey = std::array<int, 3>;
-  // std::unordered_map<VoxelKey, size_t, VoxelKeyHash, VoxelKeyEqual> voxel_map;
-  robin_hood::unordered_map<VoxelKey, size_t, VoxelKeyHash, VoxelKeyEqual> voxel_map;
-
-  voxel_map.reserve(input->data.size() / input->point_step);
+using VoxelKey = std::array<int, 3>;
+using PointIndexHashMap = robin_hood::unordered_map<VoxelKey, size_t, VoxelKeyHash, VoxelKeyEqual>;
 
+void extract_unique_voxel_point_indices(
+  const sensor_msgs::msg::PointCloud2::ConstSharedPtr & input, const VoxelSize & voxel_size,
```

Comment:
This function does not have anything to do with ownership, so a `const&` to the pointcloud is more idiomatic.
```suggestion
  const sensor_msgs::msg::PointCloud2 & input, const VoxelSize & voxel_size,
```